DeviceCode/pal/lwip_1_4_1_os/SocketsDriver/lwIP__Sockets.cpp

```diff
@@ -562,13 +562,18 @@ int LWIP_SOCKETS_Driver::SetSockOpt( SOCK_SOCKET socket, int level, int optname,
             nativeOptionName = GetNativeSockOption(optname);            
 
             switch(optname)
-            {
+            {        
+                // LINGER and DONTLINGER are not implemented in LWIP
+                case SOCK_SOCKO_LINGER:
+                case SOCK_SOCKO_DONTLINGER:
+                    errno = SOCK_ENOPROTOOPT;
+                    return SOCK_SOCKET_ERROR;

```

Comment:
Maybe I didn't get your point but the error code is pretty specific. The error returned to the user is SOCK_ENOPROTOOPT which means "Bad protocol option: An unknown, invalid or unsupported option or level was specified in a getsockopt or setsockopt call." Unfortunately, it is not possible to return a error message to the user.